pkgs/by-name/mu/mujoco/package.nix

```diff
@@ -35,8 +35,8 @@ let
     eigen3 = fetchFromGitLab {
       owner = "libeigen";
       repo = "eigen";
-      rev = "f33af052e0e60d4aa367328e7d9dffc9dedca6d8";
-      hash = "sha256-93I6MFIZ8tvdwTmiMihOaVVCdkWOTvXWZ5vYXzsMP+Q=";
+      rev = "b396a6fbb2e173f52edb3360485dedf3389ef830";
+      hash = "sha256-UroGjERR5TW9KbyLwR/NBpytXrW1tHfu6ZvQPngROq4=";
```

Comment:
Can we please test if this can be removed an favour if `pkgs.eigen3` and `find_package`? We should push to have less of this ad hoc stuff

Reply:
May I also ask to open respective `find_package` ("dependency injection") issues upstream if it's too hard to get this to work now

Reply:
> We should push to have less of this ad hoc stuff

I'd usually agree, but see https://github.com/google-deepmind/mujoco/issues/1137#issuecomment-1781811178.
We have already diverged, however:
https://github.com/NixOS/nixpkgs/blob/020268edd288f64f6808b2764cb7e50573585210/pkgs/by-name/mu/mujoco/package.nix#L120-L122

But we can remove the manual bumps with cmake: https://github.com/NixOS/nixpkgs/issues/317947#issuecomment-2155218165.

Reply:
It doesn't let me provide `eigen` directly. I would need to patch the `cmake`.
We already talked about better managing those dependencies with @samuela and @tmplt  but I fear that none of us have the time currently...

Reply:
> I'd usually agree, but see https://github.com/google-deepmind/mujoco/issues/1137#issuecomment-1781811178.

That is a common argument but it's [hard to support](https://discourse.nixos.org/t/pre-rfc-gradual-transition-of-nixos-x86-64-baseline-to-x86-64-v3-with-an-intermediate-step-to-x86-64-v2/35924/59) (i.e. we might be getting dramatic but statistically insignificant differences in performance just by using different toolchains already, and we're not even measuring) and missing the context that in nixpkgs the output binaries are not the product...

> none of us have the time currently...

yeah...